### PR TITLE
RP2: Enable bluetooth support and add Nimble port.

### PR DESCRIPTION
**Pull request by iabdalkader**

### CODE REVIEW

**Comment by iabdalkader** [comment]
I think maybe it was a bad idea to include a change to extmod in the same PR because it triggers a build of all ports. Should I move that to a separate PR ? I need to add a `mp_bluetooth_hci_uart_any` to mpbthci.h so that controller drivers (in /drivers) can read, write and poll uart without having to use a specific uart driver.

**Comment by iabdalkader** [comment]
@dpgeorge Can I get feedback on this PR ? Would you prefer to keep this BT HCI driver in drivers or a board dir  ? Also I have another PR queued up to add networking support to RP2 port, and the WiFi driver for the same module.

File: ports/rp2/mpbthciport.c [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+#define debug_printf(...) // mp_printf(&mp_plat_print, "mpbthciport.c: " __VA_ARGS__)
+#define error_printf(...) mp_printf(&mp_plat_print, "mpbthciport.c: " __VA_ARGS__)
+
+// Poll timer ID.
+static alarm_id_t poll_timer_id = 0;
+
+uint8_t mp_bluetooth_hci_cmd_buf[4 + 256];
+
+#if MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS
```
Comment (dpgeorge) [inline ports/rp2/mpbthciport.c:46]: I think we should only support the case where `MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS` is enabled.  Disabling this was a legacy thing.

File: ports/rp2/mpbthciport.c [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    poll_timer_id = 0;
+    return 0;
+}
+
+int mp_bluetooth_hci_uart_set_baudrate(uint32_t baudrate) {
+    debug_printf("mp_bluetooth_hci_uart_set_baudrate(%lu)\n", baudrate);
+    return 0;
+}
+
+int mp_bluetooth_hci_uart_any() {
```
Comment (dpgeorge) [inline ports/rp2/mpbthciport.c:144]: `void` in parenthesis

**Comment by dpgeorge** [comment]
> Would you prefer to keep this BT HCI driver in drivers or a board dir ?

I'm not sure what you mean.  What's included in this PR now is nice and clean, there are no drivers.

**Comment by iabdalkader** [comment]
> I'm not sure what you mean. What's included in this PR now is nice and clean, there are no drivers.

I'm not sure either, I think at some point this also included the WiFi/BT module (Nina) BT HCI driver, then I moved it to another PR.

**Review by dpgeorge — APPROVED** [review]
This looks good to go.

**Comment by dpgeorge** [comment]
Merged in 80f2c794e60f59179cf530d3add53a4acbe75e55 through c973cfd2f33175cacefd0608853d0db1d7fa813c

**Pull request closed without merging**